cmsl1t/plotting/rates.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import division
+import numpy as np
+from cmsl1t.plotting.base import BasePlotter
+from cmsl1t.hist.hist_collection import HistogramCollection
+from cmsl1t.hist.factory import HistFactory
+import cmsl1t.hist.binning as bn
+from cmsl1t.utils.draw import draw, label_canvas
+from cmsl1t.recalc.resolution import get_resolution_function
+
+from rootpy.context import preserve_current_style
+from rootpy.plotting import Legend
+
+
+def get_cumulative_hist(hist):
+    h = hist.clone(hist.name + '_cumul')
+    arr = np.cumsum(_reverse([bin.value for bin in hist]))
```

Comment:
Not in this PR at least ;).

Reply:
```python
def cumulative_sum(hist):
    '''
        Takes a histogram and returns an array of cumulative sums of it with
        the total first.
        E.g. histogram entries: [1, 2, 3, 4]
        Output: [10, 9, 7, 4]
    '''
    hist_values = [b.value for b in hist]
    reversed_hist = np.array(np.flipud(hist_values))
    cumsum = np.cumsum(reversed_hist)
    reversed_cumsum = np.flipud(cumsum)
    return reversed_cumsum
```

Do you expect the out

Reply:
Yeah, this looks good to me.  Might even be worth putting this function in a utility module since it could be useful in multiple places.  Or even submit a PR to rootpy itself for a new Hist1D method?

Reply:
&rarr; `cmsl1t.utils.hist`